src/vss_tools/vspec/vssexporters/vss2samm/helpers/ttl_builder_helper.py

```diff
@@ -0,0 +1,653 @@
+# Copyright (c) 2024 Contributors to COVESA
+#
+# This program and the accompanying materials are made available under the
+# terms of the Mozilla Public License 2.0 which is available at
+# https://www.mozilla.org/en-US/MPL/2.0/
+#
+# SPDX-License-Identifier: MPL-2.0
+
+from typing import Sequence
+
+from rdflib import Graph, Literal, URIRef
+from rdflib.namespace import RDF, XSD
+from vss_tools import log
+from vss_tools.vspec.tree import VSSNode
+
+from . import vss_helper as vss_helper
+from .data_types_and_units import DataTypes
+from .namespaces import Namespaces, get_node_name_from_vspec_uri, get_vspec_uri
+from .samm_concepts import SammCConcepts, SammConcepts, VSSConcepts
+from .string_helper import str_camel_case_split, str_to_lc_first_camel_case, str_to_uc_first_camel_case
+
+#
+# Builder helper, which provides a set of functions, to set up a TTL Graph,
+# build various graph nodes like property, characteristic, entity etc. and add them to this graph,
+# which then can be stored in an ESMF Aspect Model Editor loadable TTL file.
+#
+
+
+def __add_node_tuple(graph: Graph, subject_uri: URIRef, predicate_uri: URIRef, object_data: Literal | URIRef) -> bool:
```

Comment:
Might this be a  candidate to be an own exporter? Like `vspec export ttl`?

Reply:
As far as I remember, there used to be a ttl exporter in the contrib folder, but I think that this is obsolete and it was generating different formatted ttls.

Reply:
Do you think it would be reasonable to separate it into an exporter that exports it correctly?
Maybe we can reduce the complexity of that exporter then a bit? Does not need to happen  now  of course...

Reply:
Like I said, there is already this [vss2ttl](https://github.com/COVESA/vss-tools/blob/master/contrib/vspec2ttl/vspec2ttl.py) which seems to do this, but when I look at its code, it seems to be a different type of ttl format.

Both helpers: ttl_helper and ttl_builder_helper are quite bound to the SAMM formatted ttls. for this reason I am not 100% how easy would be to make these more generic so they can be used for more general purpose and why not as a stand alone ttl exporter.

Maybe as another improvement, we can think about such refactoring as further improvement.

By the way, just to add, I reused the try/catch from there :)